packages/admin-panel/src/VizBuilderApp/context/JsonEditor.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
+ */
+
+import PropTypes from 'prop-types';
+import React, { createContext, useContext, useState } from 'react';
+
+const JsonEditorContext = createContext(null);
+
+export const JsonEditorProvider = ({ children }) => {
+  const [jsonToggleEnabled, setJsonToggleEnabled] = useState(false);
+
+  return (
+    <JsonEditorContext.Provider value={{ jsonToggleEnabled, setJsonToggleEnabled }}>
+      {children}
+    </JsonEditorContext.Provider>
+  );
+};
+
+export const useJsonEditor = () => useContext(JsonEditorContext);
```

Comment:
I don't love useJsonEditor as the name here, as all this is is the toggle. It is unrelated to the actual json editor. What I mean is, if the actual json editor was exposed as a context, you could do stuff like `{ currentValue, isValid } = useJsonEditor()`. What you actually have here is more like `{ jsonToggleEnabled } = useJsonToggle()`.

Reply:
That makes sense. in case we need more global state for the tab panel in the future, I will change the name to useTabPanel().